internal/cluster/cluster.go

```diff
@@ -1833,3 +1837,25 @@ func (m *Manager) UpdateFinalizingStage(ctx context.Context, clusterID strfmt.UU
 func (m *Manager) GetHostCountByRole(clusterID strfmt.UUID, role models.HostRole, suggested bool) (*int64, error) {
 	return common.GetHostCountByRole(m.db, clusterID, role, suggested)
 }
+
+func (m *Manager) HasBaseCapabilities(capabilities *installcfg.Capabilities, baselineCapability configv1.ClusterVersionCapabilitySet) bool {
```

Comment:
WDYT if we include logic of retrieving capabilities from the host? That way from a user perspective we can forget about that, and we can check capabilities on a cluster:
```

c := getMyCluster()
if cluster.HasBaseCapabilities(c, baselineCap) && cluster.HasAdditionalCapabilities(c, includes) && !cluster.HasAdditionalCapabilities(c, excludes) {
...
}
```
Wouldn't something like this keep implementation simple and the interface clearer? WDYT?